app/assets/javascripts/application.js

```diff
@@ -22,6 +22,65 @@
 //= require_tree ./users
 //= require_tree ./utils
 
-$('body').on('click', '[disabled], .disabled', function(e) {
+$('body').on('click', '[disabled], .disabled', function (e) {
   e.preventDefault();
 });
+
+$(document).ready(function () {
+
+  // Elements specific to LDAP Connector page; check for existence before changing
+  if ($("#ldap_conn_save").length > 0 && $("#ldap_conn_message").length > 0) {
+    $("#ldap_conn_save").prop("disabled", true);
+    $("#ldap_conn_message")[0].style.color = "black";
+  };
+
+  // Run LDAP Connection Test with button click
+  $("#ldap_conn_test").click(function (event) {
+
+    // Get values from form 
+    var host = $('#dex_connector_ldap_host').val();
+    var port = $('#dex_connector_ldap_port').val();
+    var start_tls = $("#dex_connector_ldap_start_tls_true").parent().hasClass('active');
+    var cert = $('#dex_connector_ldap_certificate').val();
+    // Flip logic of #anonymous_bind_toggle since expanded div means anonymous binding is 'false'
+    var anon_bind = ($('#anonymous_bind_toggle').attr('aria-expanded') == "false");
+    var dN = $('#dex_connector_ldap_bind_dn').val();
```

Comment:
Is there a reason the N here in dN is capital?

Reply:
I was attempting to keep it as the capitalized LDAP convention, 'DN'.  I settled with 'dN' because Ruby automatically sets capitalized variables as constants.  There is no reason to not follow ruby convention and change to 'dn' though.  So I am changing it to 'dn'.